ESP32_AP-Flasher/src/web.cpp

```diff
@@ -241,12 +260,85 @@ uint8_t wsClientCount() {
     return ws.count();
 }
 
+
+#ifdef W5500_ETH
+	void Ethernet_event(WiFiEvent_t event)
+	{
+	  switch (event)
+	  {
+	    case ARDUINO_EVENT_ETH_START:
+	      AWS_LOG(F("\nETH Started"));
+
+	      uint8_t  address[6];
+	      WiFi.macAddress(address);
+	      char hexmac1[40];
+	      sprintf(hexmac1, "openepaperlink-%02X%02X%02X", address[3],address[4], address[5]);
+	      ETH.setHostname(hexmac1);
+	      break;
+
+	    case ARDUINO_EVENT_ETH_CONNECTED:
+	      AWS_LOG(F("ETH Connected"));
+	      break;
+
+	    case ARDUINO_EVENT_ETH_GOT_IP:
+	      if (!ESP32_W5500_eth_connected)
+	      {
+	        AWS_LOG3(F("ETH MAC: "), ETH.macAddress(), F(", IPv4: "), ETH.localIP());
+
+	        if (ETH.fullDuplex())
+	        {
+	          AWS_LOG0(F("FULL_DUPLEX, "));
+	        }
+	        else
+	        {
+	          AWS_LOG0(F("HALF_DUPLEX, "));
+	        }
+
+	        AWS_LOG1(ETH.linkSpeed(), F("Mbps"));
+
+	        ESP32_W5500_eth_connected = true;
+	      }
+
+	      break;
+
+	    case ARDUINO_EVENT_ETH_DISCONNECTED:
+	      AWS_LOG("ETH Disconnected");
+	      ESP32_W5500_eth_connected = false;
+	      break;
+
+	    case ARDUINO_EVENT_ETH_STOP:
+	      AWS_LOG("ETH Stopped");
+	      ESP32_W5500_eth_connected = false;
+	      break;
+
+	    default:
+	      break;
+	  }
+	}
+#endif
+
 void init_web() {
     wsMutex = xSemaphoreCreateMutex();
-    WiFi.mode(WIFI_STA);
-    WiFi.setTxPower(static_cast<wifi_power_t>(config.wifiPower));
+	
+    #ifndef W5500_ETH
+        WiFi.mode(WIFI_STA);
+        WiFi.setTxPower(static_cast<wifi_power_t>(config.wifiPower));
+
+        wm.connectToWifi();
+    #else	
+	    uint8_t  address[6];
+	    WiFi.macAddress(address);
+
+	    WiFi.onEvent(Ethernet_event);
+
+	    ETH.begin( ETH_MISO_GPIO, ETH_MOSI_GPIO, ETH_SCK_GPIO, ETH_CS_GPIO, ETH_INT_GPIO, SPI_CLOCK_MHZ, ETH_SPI_HOST, address );
+
+	    ESP32_W5500_waitForConnect();
+
+		vTaskDelay(1000 / portTICK_PERIOD_MS);
 
-    wm.connectToWifi();
+	    init_udp();
```

Comment:
init_udp() should also be done on wifi

Reply:
no, init_udp is called in the wifi_event in the wifimanager.  (which is no longer called for the ethernet stuff)

Reply:
ah, ok!